flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/inference/TypeInferenceOperandInference.java

```diff
@@ -81,7 +80,7 @@ public void inferOperandTypes(
                     false)) {
                 inferOperandTypesOrError(unwrapTypeFactory(callBinding), callContext, operandTypes);
             }
-        } catch (ValidationException | CalciteContextException e) {
+        } catch (ValidationException e) {
             // let operand checker fail
```

Comment:
rethrowing here as `ValidationException` will not help for complex cases and will lead again to `StackOverflow`: first it rethrows as `ValidationException` and then swallows.

For that reason there is `testNestedCoalesceOnInvalidField` highlighting this case with  a query
```sql
SELECT coalesce(SELECT coalesce(SELECT coalesce(invalid)))
```